pkg/autoupdate/tuf/library_lookup.go

```diff
@@ -0,0 +1,75 @@
+package tuf
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/kolide/launcher/pkg/autoupdate"
+)
+
+// CheckOutLatest returns the path to the latest downloaded executable for our binary, as well
+// as its version.
+func CheckOutLatest(binary autoupdatableBinary, rootDirectory string, updateDirectory string, channel string) (string, string, error) {
+	if updateDirectory == "" {
+		updateDirectory = defaultLibraryDirectory(rootDirectory)
+	}
+
+	releasePath, releaseVersion, err := findExecutableFromRelease(binary, LocalTufDirectory(rootDirectory), channel, updateDirectory)
+	if err == nil {
+		return releasePath, releaseVersion, nil
+	}
+
+	// If we can't find the specific release version that we should be on, then just return the executable
+	// with the most recent version in the library
+	return mostRecentVersion(binary, updateDirectory)
+}
+
+// findExecutableFromRelease looks at our local TUF repository to find the release for our
+// given channel. If it's already downloaded, then we return its path and version.
+func findExecutableFromRelease(binary autoupdatableBinary, tufRepositoryLocation string, channel string, baseUpdateDirectory string) (string, string, error) {
+	// Initialize a read-only TUF metadata client to parse the data we already have downloaded about releases.
+	metadataClient, err := readOnlyTufMetadataClient(tufRepositoryLocation)
+	if err != nil {
+		return "", "", errors.New("could not initialize TUF client, cannot find release")
+	}
```

Comment:
What's the behavior here if there's no local repo? (eg: first boot)

Reply:
If we can't read from the TUF repo for whatever reason, then we'd fall back to picking the most recent version in the update library. If there are no updates in the update library either, then we'd assume that the current running executable (likely the installed version, at this point) is the correct one to be running.

Reply:
Perfect. I'm getting turned around because there are many functions that all call each other so tracing the flow isn't super clear to me.